src/telemdaemon.c

```diff
@@ -335,82 +285,14 @@ void process_record(TelemDaemon *daemon, client *cl)
         /* TODO : check if the body is within the limits. */
         body = msg + header_size;
 
-        /* Save record in journal */
-        save_entry_to_journal(daemon, temp, headers);
-
-        /* Save local copy */
-        if (daemon->record_retention_enabled) {
-                save_local_copy(daemon, body);
-        }
-
-        /* Bail out if server delivery is not enabled */
-        if (!daemon->record_server_delivery_enabled) {
-#ifdef DEBUG
-                fprintf(stdout, "process_record: record server delivery disabled\n");
-#endif
-                goto end;
-        }
-
-        if (inside_direct_spool_window(daemon, time(NULL))) {
-                telem_log(LOG_INFO, "process_record: delivering directly to spool\n");
-                spool_record(daemon, headers, body);
-                goto end;
-        }
-        if (daemon->record_window_length == -1 ||
-            daemon->byte_window_length == -1) {
-                telem_log(LOG_ERR, "Invalid value for window length\n");
+        /* Save record to stage */
+        ret = asprintf(&recordpath, "%s/XXXXXX", DEFAULT_STAGE_DIR);
```

Comment:
Your understanding is correct. I agree that adding a staging folder might not have helped simplify the code which is the original intent of the refactoring and I'm looking at reusing spool directory. I used an staging directory, because I thought it would be simpler. Implementation simpler was simpler, however the architecture got more complex.

The ultimate goal is for telemprobd to handle clients as fast as possible and that's for future work. However, this work has to be performed first to get there.